packages/timelock-manager/contracts/TimeLockManagerReversible.sol

```diff
@@ -0,0 +1,255 @@
+/// TimelockManager to timelock them. These tokens will then be vested to their
+/// recipient linearly, starting from releaseStart and ending at releaseEnd of
+/// the respective timelock.
+
+contract TimelockManagerReversible is Ownable, ITimelockManagerReversible {
+    using SafeMath for uint256;
+
+    struct Timelock {
+        uint256 totalAmount;
+        uint256 remainingAmount;
+        uint256 releaseStart;
+        uint256 releaseEnd;
+    }
+
+    IApi3Token public immutable api3Token;
+    mapping(address => Timelock) public timelocks;
+
+    /// @param api3TokenAddress Address of the API3 token contract
+    /// @param timelockManagerOwner Address that will receive the ownership of
+    /// the TimelockManager contract
+    constructor(
+        address api3TokenAddress, 
+        address timelockManagerOwner
+        ) 
+        public 
+    {
+        api3Token = IApi3Token(api3TokenAddress);
+        transferOwnership(timelockManagerOwner);
+    }
+
+    /// @notice Called by the ContractOwner to stop the vesting of
+    /// a recipient
+    /// @param recipient Original recipient of tokens
+    /// @param destination Destination of the excess tokens vested to the addresss
+    function stopVesting(
+        address recipient, 
+        address destination
+        )
+        external
+        override
+        onlyOwner
+        onlyIfRecipientHasRemainingTokens(recipient)
+    {
+        uint256 withdrawable = getWithdrawable(recipient);
+        uint256 reclaimedTokens =
+            timelocks[recipient].remainingAmount.sub(withdrawable);
+        timelocks[recipient].remainingAmount = withdrawable;
+        timelocks[recipient].releaseEnd = now;
+        require(
+            api3Token.transfer(destination, reclaimedTokens),
+            "API3 token transfer failed"
+        );
+        emit StoppedVesting(recipient, destination, reclaimedTokens);
+    }
+
+    /// @notice Transfers API3 tokens to this contract and timelocks them
+    /// @dev source needs to approve() this contract to transfer amount number
+    /// of tokens beforehand.
+    /// A recipient cannot have multiple timelocks.
+    /// @param source Source of tokens
+    /// @param recipient Recipient of tokens
+    /// @param amount Amount of tokens
+    /// @param releaseStart Start of release time
+    /// @param releaseEnd End of release time
+    function transferAndLock(
+        address source,
+        address recipient,
+        uint256 amount,
+        uint256 releaseStart,
+        uint256 releaseEnd
+        ) 
+        public 
+        override 
+        onlyOwner
+    {
+        require(
+            timelocks[recipient].remainingAmount == 0,
+            "Recipient has remaining tokens"
+        );
+        require(amount != 0, "Amount cannot be 0");
+        require(
+            releaseEnd > releaseStart,
+            "releaseEnd not larger than releaseStart"
+        );
+        require(releaseStart > now, "releaseStart not in the future");
```

Comment:
Can you remove this require and update the test accordingly please? This will give us a bit more flexibility regarding how we can use the contract.